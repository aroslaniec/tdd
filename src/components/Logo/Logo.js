import React, { Component } from "react";
import PropTypes from "prop-types";

import "./Logo.scss";
import { APP_NAME } from "../../constants";

class Logo extends Component {
  static propTypes = {
    pathToSrc: PropTypes.string.isRequired,
    title: PropTypes.string
  };
  static defaultProps = {
    title: APP_NAME
  };
  static displayName = "Logo";

  render() {
    return (
      <div className="logo">
        <div className="logo__wrapper">
          <img src={this.props.pathToSrc} alt={this.props.title} />
        </div>
      </div>
    );
  }
}

export default Logo;
