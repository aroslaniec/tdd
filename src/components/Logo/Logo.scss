.logo__wrapper {
  width: 100%;
  height: 100%;

  img {
    animation: logo-spin infinite 20s linear;
    max-width: 100%;
    max-height: 100%;
    display: block;
    width: 100%;
    object-fit: contain;
  }
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
