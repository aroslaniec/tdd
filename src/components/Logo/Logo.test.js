import React from "react";
import { shallow } from "enzyme";

import Logo from "./Logo";

describe("Logo", () => {
  let MOCKED_PROPS = {};

  beforeEach(() => {
    MOCKED_PROPS = {
      pathToSrc: "./lorem-ipsum.png"
    };
  });

  it("should render correctly", () => {
    const wrapper = shallow(<Logo {...MOCKED_PROPS} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should have className set accordingly to naming convention", () => {
    const wrapper = shallow(<Logo {...MOCKED_PROPS} />);

    expect(wrapper.hasClass("logo")).toBeTruthy();
  });
});
