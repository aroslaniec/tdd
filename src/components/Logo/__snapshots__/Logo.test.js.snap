// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render correctly 1`] = `
<div
  className="logo"
>
  <div
    className="logo__wrapper"
  >
    <img
      alt="My awesome TDD App!"
      src="./lorem-ipsum.png"
    />
  </div>
</div>
`;
